R/call_utils.R

```diff
@@ -49,108 +21,96 @@ call_check_parse_varname <- function(varname) {
 #'   }
 #'
 #' @examples
-#' teal.slice:::call_condition_choice("SEX", choices = c(1, 2))
-#' teal.slice:::call_condition_choice(as.name("SEX"), choices = "F")
-#' teal.slice:::call_condition_choice("SEX", choices = c("F", "M"))
-#' teal.slice:::call_condition_choice("SEX", choices = factor(c("F", "M")))
-#' teal.slice:::call_condition_choice("x$SEX", choices = Sys.Date())
-#' teal.slice:::call_condition_choice("trunc(x$SEX)", choices = Sys.time())
+#' \dontrun{
+#' call_condition_choice("SEX", choices = c(1, 2))
+#' call_condition_choice("SEX", choices = "F")
+#' call_condition_choice("SEX", choices = c("F", "M"))
+#' call_condition_choice("x$SEX", choices = Sys.Date())
+#' call_condition_choice("x$SEX", choices = Sys.time())
+#' }
 #' @return a `call`
 #' @keywords internal
 call_condition_choice <- function(varname, choices) {
-  varname <- call_check_parse_varname(varname)
-
-  if (is.factor(choices)) {
-    choices <- as.character(choices)
-  } else if (inherits(choices, "Date")) {
-    choices <- format(choices)
-  } else if (inherits(choices, c("POSIXct", "POSIXlt"))) {
-    choices <- format(choices)
-  }
-
+  checkmate::assert_string(varname)
+  checkmate::assert_multi_class(
+    choices,
+    c("integer", "numeric", "character", "factor", "Date", "POSIXct", "POSIXlt")
+  )
```

Comment:
Should allow all types if, according to documentation of `FilterStateChoices` states that it is only used for characters and factors?

Reply:
Should allow more than `character` and `factor`. `ChoicesFilterState` can be based on more than just that.

https://github.com/insightsengineering/teal.slice/blob/1128a868f4fe4934409bf60adca0da3d68e46ec1/R/FilterState-utils.R#L128

In the above line I referenced - I'm not sure whether it should be applied to all data types actually. If we had just 5-dates in a data it doesn't make sense to give a calendar to select dates from.

Reply:
Well, currently `ChoicesFilterStates` is created only for characters, factors and numerics. Do we expand to other types?

Reply:
Yup, I think we should expand to other types as well.